### PR TITLE
Infinite recursive loop issue fixed

### PR DESCRIPTION
**Pull request by jeet1293**
Infinite recursive loop issue fixed

### CODE REVIEW

**Comment by jeet1293** [comment]
hello @cloudcogsio, can you please review this PR and merge if it looks good to you?

**Comment by cloudcogsio** [comment]
> hello @cloudcogsio, can you please review this PR and merge if it looks good to you?

Hi, can you write up the issue you encountered along with how I can reproduce so I can easily check and confirm? Thanks.

**Comment by jeet1293** [comment]
don’t have a code sample to share, but I can explain what I was working on when I encountered this issue (June 2024).

I was building a Laravel 8 application and had integrated your package into it. I tried both code snippets provided in your README file, but neither returned a response. After reviewing your code, I removed one specific line, and then I was able to get a response.

1. **Generate a new Laravel 8 project**:

    ```bash
    composer create-project --prefer-dist laravel/laravel="8.*" "FAC Demo"
    ```

2. **Navigate to the project directory**:

    ```bash
    cd "FAC Demo"
    ```

3. **Install the `cloudcogsio/omnipay-firstatlanticcommerce-gateway` package**:

    ```bash
    composer require cloudcogsio/omnipay-firstatlanticcommerce-gateway
    ```

File: src/FACGateway.php [hunk 1]
```diff
@@ -83,7 +83,6 @@ public function getDefaultParameters()
      */
     public function setMerchantResponseURL($url)
     {
-        $this->setReturnUrl($url);
```
Comment (cloudcogsio) [inline src/FACGateway.php:86]: Add return here

File: src/FACGateway.php [hunk 2]
```diff
@@ -83,7 +83,6 @@ public function getDefaultParameters()
      */
     public function setMerchantResponseURL($url)
     {
-        $this->setReturnUrl($url);
         return $this->setParameter(Constants::CONFIG_KEY_MERCHANT_RESPONSE_URL, $url);
```
Comment (cloudcogsio) [inline src/FACGateway.php:86]: Remove

**Review by cloudcogsio — CHANGES_REQUESTED** [review]
In `setMerchantResponseURL`, remove
`return $this->setParameter(Constants::CONFIG_KEY_MERCHANT_RESPONSE_URL, $url);`
and add
`return $this->setReturnUrl($url);`

Then in `setReturnUrl`, add
`$this->setParameter(Constants::CONFIG_KEY_MERCHANT_RESPONSE_URL, $url);` before the current `return` statement,

Both functions should look like,

```
public function setMerchantResponseURL($url)
    {
        return $this->setReturnUrl($url);
    }
```
```
public function setReturnUrl($url)
    {
        $this->setParameter(Constants::CONFIG_KEY_MERCHANT_RESPONSE_URL, $url);
        return $this->setParameter("returnUrl", $url);
    }
```

Let me know if that works in your project. You can make the changes and resubmit PR.

**Review by cloudcogsio — APPROVED** [review]

**Pull request merged**